app/src/main/java/com/hedvig/app/feature/offer/OfferActivity.kt

```diff
@@ -52,17 +48,33 @@ class OfferActivity : BaseActivity(R.layout.activity_offer) {
     private val offerViewModel: OfferViewModel by viewModel()
     private val tracker: OfferTracker by inject()
 
+    private val animationHandler = Handler()
+    private var hasTriggeredAnimations = false
+    private var lastAnimationHasCompleted = false
+
     private lateinit var factAreaBinder: FactAreaBinder
     private lateinit var termsBinder: TermsBinder
+    private lateinit var perilBinder: PerilBinder
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         factAreaBinder = FactAreaBinder(offerFactArea as LinearLayout)
         termsBinder = TermsBinder(offerTermsArea as LinearLayout, tracker)
+        perilBinder = PerilBinder(offerPerilArea as LinearLayout, supportFragmentManager)
 
         offerViewModel.data.observe(lifecycleOwner = this) {
             it?.let { data ->
+                /* Bind perils */
+                val perils = data.lastQuoteOfMember.asCompleteQuote?.perils.orEmpty()
+                perilBinder.bind(perils, data)
```

Comment:
It should be sufficient to send `data` to this function as `data` contains the perils